src/ipc/ipc4/logging.c

```diff
@@ -151,9 +151,9 @@ int ipc4_logging_enable_logs(bool first_block,
 				       &ops, NULL, MTRACE_IPC_CORE, 0);
 		schedule_task(&mtrace_task, mtrace_aging_timer * 1000, 0);
 	} else  {
+		schedule_task_cancel(&mtrace_task);
```

Comment:
@lyakh Did you actually hit this? The mtrace_task is not directly dependent on the backend or adsp_mtrace trace, so ordering should not matter.

Reply:
@kv2019i it was one of the measures that I had to take to fix the pipeline tearing-down path, but in the end I'm not sure if it was necessary, maybe the decisive fixes were different. I can re-test without this

Reply:
Ack @lyakh . The updated code is not wrong either, just curious whether my lifetime streak of bugfree code has come to an end now. So if you have CI passing et al, you can keep this as well.

Reply:
ok, I've pushed #9290 - we'll see whether this is optional